### DISCUSSION TITLE
pfsense-api settings after pfsense upgrade.

### DISCUSSION THREAD

**Discussion by Eryniox** [jaredhendrickson13/pfsense-api#389] [Q&A] [answered]
Hi.
Do you have any suggestions for the best way to keep the settings for pfsense-api after upgrading pfsense?
What I mean is:
Last time I upgraded pfsense (to 2.6, I think), all settings (including api tokens for all users) were deleted from xml-file by pfsense updater.
Then I had to restore the pfsense configuration to the one from before upgrading. And had to do it twice (since it failed the first time). And each time it reinstalled all packages, so it took a while 😄 

I tried to see if I could only restore the pfsense-api settings, but failed.

Is this the normal way to do this? Or just recreate settings and tokens?

**Comment by jaredhendrickson13** [comment, marked as answer]
Hey!

By default the API package will store a persistent backup of the API configuration at `/usr/local/share/pfSense-pkg-API/backup.json` (as long as you haven't disabled the 'Persistent configuration' setting in the API configuration). Whenever the package is reinstalled it looks for a backup in this location and will automatically attempt to restore the configuration from that file. I'd recommend keeping a copy of the backup.json file off of your pfSense systems before you update just in case. You can manually restore from the backup file by running `pfsense-api restore` from the command line or manually take a backup by running `pfsense-api backup`.

Hope this helps!

**Reply by Eryniox** [reply]
Great, thank you! I'll try this when upgrading to 2.7 😄
